lib/models/offline_recorder.dart

```diff
@@ -146,25 +147,17 @@ class OfflineRecorder {
     }
   }
 
-  /// Gets the news of the given date, throws FileSystemException if the file containing the date's news doesn't exist
-  Future<News> loadNews(DateTime date) async {
-    // Sanitize inputs
-    String day = date.day < 10 ? "0${date.day}" : date.day.toString();
-    String month =
-        date.month < 10 ? "0${date.month.toString()}" : date.month.toString();
-    if (date.year < 1000) {
-      throw UnimplementedError("The date is too far in the past");
-    }
-
-    String filePath =
-        _appOfflineNewsPath + "${date.year}-$month-${day}_transcript.json";
-    if (!await File(filePath).exists()) {
-      throw FileSystemException("$filePath doesn't exist");
+  Future<News?> _readNewsFile(File file) async {
+    if (!(file.path
+        .substring(file.path.length - "XXXX-XX-XX_transcript.json".length)
+        .startsWith(RegExp(r"[0-9]{4}-[0-9]{2}-[0-9]{2}")))) {
+      return Future(() => null);

```

Comment:
```suggestion
      return null;
```

Can be simplified since the function is marked async.